test/test_dht_storage.cpp

```diff
@@ -177,4 +184,48 @@ TORRENT_TEST(dht_storage_counters)
 	TEST_EQUAL(s->counters().mutable_data, 1);
 }
 
+TORRENT_TEST(dht_storage_set_custom)
+{
+	settings_pack p;
+	p.set_bool(settings_pack::enable_dht, false);
+	lt::session ses(p);
+
+	bool r = ses.is_dht_running();
+	TEST_CHECK(!r);
+
+	r = ses.set_dht_storage(dht_custom_storage_constructor);
+	TEST_CHECK(r);
+
+	p.set_bool(settings_pack::enable_dht, true);
+	ses.apply_settings(p); // async with dispatch
+	r = ses.is_dht_running();
+	TEST_CHECK(r);
```

Comment:
shouldn't there be a check that the custom storage class was instantiated or something?